agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -1,164 +1,90 @@
+ * a replacement pair consisting of methods, the original method is replaced by the replacement
+ * method. Replacement methods must be static. If the pair consists of classes, the methods of the
+ * original class will be replaced by identically named static methods of a class with the
+ * replacement name. (If the original method is non-static, the class
+ *
+ * @see MapCallsAgent
  */
 public class CallReplacementTransformer implements ClassFileTransformer {
 
-  boolean debug = false;
-  boolean log_on = false;
-  boolean debug_class = false;
-
-  /** current Constant Pool */
-  static ConstantPoolGen pgen = null;
-
-  protected InstructionFactory ifact;
-
+  //XXX fix path construction for windows
   /** Debug information about which classes are transformed and why */
-  public static SimpleLog debug_transform = new SimpleLog(false);
+  private static SimpleLog debug_transform =
+      new SimpleLog(MapCallsAgent.debugPath + "/transform-log.txt", MapCallsAgent.debug);
 
-  public static SimpleLog debug_instrument = new SimpleLog(false);
-  public static SimpleLog debug_instrument_inst = new SimpleLog(false);
+  private static SimpleLog debug_instrument_inst =
+      new SimpleLog(MapCallsAgent.debugPath + "/instrument-log.txt", MapCallsAgent.debug);
 
   /** Debug information on method maping */
-  public static SimpleLog debug_map = new SimpleLog("method_mapping.txt", true);
-
-  /** Class that defines a method (by its fully-qualified name and argument types) */
-  static class MethodDef {
-    /** Fully-qualified method name, such as "javax.swing.JOptionPane.showMessageDialog" */
-    String name;
-    /** The argument types */
-    Type[] arg_types;
-
-    MethodDef(String name, Type[] arg_types) {
-      this.name = name;
-      this.arg_types = arg_types;
-    }
+  private static SimpleLog debug_map =
+      new SimpleLog(MapCallsAgent.debugPath + "/method_mapping.txt", MapCallsAgent.debug);
 
-    boolean equals(String name, Type[] arg_types) {
-      if (!name.equals(this.name)) return false;
-      if (this.arg_types.length != arg_types.length) return false;
-      for (int ii = 0; ii < arg_types.length; ii++)
-        if (!arg_types[ii].equals(this.arg_types[ii])) return (false);
-      return (true);
-    }
+  /** Map from a method to its replacement. */
+  private final ConcurrentHashMap<MethodDef, MethodDef> replacementMap;
 
-    @Override
-    public boolean equals(Object obj) {
-      if (!(obj instanceof MethodDef)) return false;
-      MethodDef md = (MethodDef) obj;
-      return equals(md.name, md.arg_types);
-    }
-
-    @Override
-    public int hashCode() {
-      int code = name.hashCode();
-      for (Type arg : arg_types) code += arg.hashCode();
-      return code;
-    }
-
-    @Override
-    public String toString() {
-      return String.format("%s (%s)", name, UtilMDE.join(arg_types, ", "));
-    }
-  }
+  /** The list of packages to exclude from transformation */
```

Comment:
Do these have a trailing period or not?  Document anywhere that they do, because those aren't really packages, but prefixes.  (Please give all such variables names like `packagePrefix` instead of `package`.)